lib/Echidna/Solidity.hs

```diff
@@ -158,6 +163,17 @@ linkLibraries [] = ""
 linkLibraries ls = "--libraries " ++
   iconcatMap (\i x -> concat [x, ":", show $ addrLibrary + toEnum i, ","]) ls
 
+filterMethods :: MonadThrow m => Filter -> NE.NonEmpty SolSignature -> m (NE.NonEmpty SolSignature)
+filterMethods (Whitelist [])  _ = throwM InvalidMethodFilters 
+filterMethods (Whitelist ic) ms = return $ NE.fromList $ NE.filter (\(m, _) -> T.unpack m `elem` ic) ms
+filterMethods (Blacklist ig) ms = return $ NE.fromList $ NE.filter (\(m, _) -> T.unpack m `notElem` ig) ms
```

Comment:
is `NonEmpty` the correct behavior here? maybe we want to also throw `InvalidMethodFilters` if the resulting list is empty

Reply:
Fixed.